smartsim/_core/launcher/dragon/dragonLauncher.py

```diff
@@ -318,13 +132,39 @@ def run(self, step: Step) -> t.Optional[str]:
         cmd = step.get_launch_cmd()
         out, err = step.get_output_files()
 
-        if isinstance(step, DragonStep):
+        if isinstance(step, DragonBatchStep):
+            if isinstance(step.batch_settings, SbatchSettings):
+                # wait for batch step to submit successfully
+                return_code, out, err = self.task_manager.start_and_wait(cmd, step.cwd)
```

Comment:
ditto on multiple branches doing the same thing. let's see if we can detect the active launcher for the step and just make use of the API on the launcher being shared.

Reply:
consider adding a `launcher.step_prefix` property so you can do something like:

```
step_id = f"{active_launcher.step_prefix}-{pbs_step_id}"
```

Reply:
I think the only information we can rely on comes from the `BatchSettings` type. I refactored the code to reduce code replication.